bin/test-files

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'faker'
+require 'fileutils'
+require 'securerandom'
+require 'tty-progressbar'
+require 'pathname'
+
+options = { segments: 0, files: 1, zip: false }
+parser = OptionParser.new do |option_parser|
+  option_parser.banner = 'Usage: bin/test-files [options]'
+  option_parser.on('-fFILES', '--files FILES', Integer, "Number of files. Default is #{options[:file]}.")
+  option_parser.on('-sSEGMENTS', '--segments SEGMENTS', Integer, "Number of directory path segments. Default is #{options[:segments]}.")
+  option_parser.on('-dDIR', '--dir DIR', String, "Directory for the files. If not will be provided, a directory name will be generated.")
+  option_parser.on('-z', '--zip', "Zip the files.")
+  option_parser.on('-h', '--help', 'Displays help.') do
+    puts option_parser
+    exit
+  end
+end
+
+parser.parse!(into: options)
+
+def write_file(path:, filename:)
+  FileUtils.mkdir_p(path) unless Dir.exist?(path)
+  filepath = File.join(path, filename)
+  File.open(filepath, 'w') do |file|
+    file.write(filename)
+  end
+end
+
+def generate_path(base_path:, segments:)
+  rand_segments = SecureRandom.random_number(0..segments)
+  return base_path if rand_segments.zero?
+
+  Faker::File.dir(segment_count: rand_segments, root: base_path)
+end
+
+def tty_progress_bar(files)
+  TTY::ProgressBar.new(
+    "Generating [:bar] (:percent (:current/:total), rate: :rate/s, mean rate: :mean_rate/s, :elapsed total, ETA: :eta_time)",
+    bar_format: :crate,
+    total: files
+  )
+end
+
+def zip(path:)
+  puts "Zipping"
+  zip_path = "#{path}.zip"
+  File.delete(zip_path) if File.exist?(zip_path)
+  system("cd #{path} && zip -qdg -r ../#{Pathname.new(path).basename}.zip *")
+  puts "Zipped to #{zip_path}"
+end
+
+base_path = options[:dir] || "test-data/test-data-f#{options[:files]}-s#{options[:segments]}"
+FileUtils.rm_rf(base_path) if Dir.exist?(base_path)
+
+progress_bar = tty_progress_bar(options[:files])
+progress_bar.start
+
+
+for i in 1..options[:files]
+  path = generate_path(base_path: base_path, segments: options[:segments])
+  filename = Faker::File.file_name(dir: '')
```

Comment:
@justinlittman there is a small limitation here where the name pool seems to be limited to 5621 items, so this is the maximum number of files that will be generated (even through the script doesn't stop, runs all the way through without error, files are just overwritten): 

```
Generating [▣▣▣▣▣▣] (100% (25000/25000), rate: 1397.23/s, mean rate: 1250.61/s, 19s total, ETA: 12:44:18)
ls -l test-data/test-data-f25000-s0 | wc -l
    5621
```

Updating this call to `Faker::File.unique.file_name(dir: '')`  results in an error:
```
Generating [▣▣▣⬚⬚⬚⬚⬚⬚⬚⬚] (23% (5824/25000), rate:  6.16/s, mean rate: 589.82/s,  9s total, ETA: 12:47:11)/Users/amcollie/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/faker-3.2.0/lib/helpers/unique_generator.rb:22:in `method_missing': Retry limit exceeded for file_name (Faker::UniqueGenerator::RetryLimitExceeded)
	from bin/test-files:66:in `block in <main>'
	from bin/test-files:64:in `each'
	from bin/test-files:64:in `<main>'
```
Note: this is in the README: https://github.com/faker-ruby/faker#ensuring-unique-values


I don't necessarily think this should hold up this PR, but should be called out in a comment or something. I'll poke at the library a bit more to see if I can offer a suggested approach.

Reply:
This may not be the best approach, but for me locally, I updated line 66 here to be:

```
filename = "/tf_#{i}_#{File.basename(Faker::File.file_name(dir: ''))}"
```